src/main/java/org/neo4j/junit/jupiter/causal_cluster/Neo4jServer.java

```diff
@@ -55,10 +65,24 @@
 	 */
 	URI getURI();
 
+	/**
+	 * @return The URI into this server but using the {@code bolt} instead of the {@code neo4j} protocol.
+	 */
 	URI getDirectBoltUri();
 
 	/**
 	 * @return True if the underlying container is running.
 	 */
 	boolean isContainerRunning();
+
+	/**
+	 * @return The type of this server.
+	 */
+	Type getType();
+
+	/**
+	 * @param database The database for which the roles should be retrieved.
+	 * @return The roles that this server has for the database in question.
+	 */
+	List<String> getRolesFor(String database);
```

Comment:
I'm not sure that we need this in this interface. It looks like quite a lot of extra work to maintain - what's the benefit to having it here rather than users that care about this implementing it themselves or getting it from some other tooling/dependency that just requires a neo4j driver for example?

Reply:
I'm gonna remove that. You're right, it's too much of an effort. Also with supporting 3.5 in mind. Thanks!